rmw_fastrtps_shared_cpp/src/subscription.cpp

```diff
@@ -47,17 +39,34 @@ destroy_subscription(
   assert(subscription->implementation_identifier == identifier);
   static_cast<void>(identifier);
 
+  // Get RMW Subscriber
   rmw_ret_t ret = RMW_RET_OK;
   auto info = static_cast<CustomSubscriberInfo *>(subscription->data);
-  if (!Domain::removeSubscriber(info->subscriber_)) {
-    RMW_SET_ERROR_MSG("failed to remove subscriber");
-    ret = RMW_RET_ERROR;
-  }
-  delete info->listener_;
 
-  Participant * participant = participant_info->participant;
-  _unregister_type(participant, info->type_support_);
-  delete info;
+  // NOTE: Topic deletion and unregister type is done in the participant
+  if (nullptr != info) {
+    // Delete DataReader
+    ReturnCode_t ret = participant_info->subscriber_->delete_datareader(info->subscriber_);
```

Comment:
Here there is an example of how confusing is to call `subscriber_` both at the `Subscriber` and the `DataReader`.